documentation/modules/exploit/multi/http/splunk_privilege_escalation_cve_2023_32707.md

```diff
@@ -0,0 +1,200 @@
+## Vulnerable Application
+
+### Description
+
+An authorization issue within Splunk Enterprise allows any user with the edit_user capability to take over
+the admin account (or any other chosen account) by simply changing its password.
+
+On June 1, 2023, Splunk released a software update that addressed this vulnerability (CVE-2023-32707).
+
+The following products are affected:
+
+Splunk Enterprise:
+ - from 8.1 before 8.1.14
+ - from 8.2 before 8.2.11
+ - from 9.0 before 9.0.5
+
+Splunk Cloud Platform:
+ - before 9.0.2303.100
+
+### Exploitation
+
+This module exploits this authorization issue to take over the admin account (or any other account with
+the capability `install_apps`) to deploy an app within Splunk, aiming to achieve remote code execution.
+
+To achieve that this module:
+- Will change the password of the targeted user;
+- Will deploy an app with a malicious payload;
+
+After the execution the cleanup method will be called and:
+- Should delete the deployed app;
+
+### Setup
+
+Create a Splunk's docker container with the following command:
+
+```bash
+    docker run --rm -p 8000:8000 -p 8089:8089 -e "SPLUNK_START_ARGS=--accept-license" -e "SPLUNK_PASSWORD=password" --name splunk-9.0.4 splunk/splunk:9.0.4
+```
+
+```bash
+    # Creating non-admin user
+    $ curl -k -u admin:password https://localhost:8089/services/authentication/users -d name=redway -d password=changeme -d roles=user -d createrole=1 -X POST
+    # Adding capability to edit_user to the non-admin role
+    $ curl -k -u admin:password https://localhost:8089/services/authorization/roles/user-redway -d capabilities=edit_user -X POST
+```
+
+:exclamation: One must log in at least once on the web interface (https://localhost:8000). Otherwise, this module will fail to get the CSRF
+token on the `appinstall`.
+
+## Verification Steps
+Follow [Setup](#setup) and [Scenarios](#scenarios).
+
+## Options
+
+### USERNAME (required)
+
+The username that with the capability `edit_user` to authenticate with.
+
+### PASSWORD (required)
+
+The password of the user to authenticate with.
+
+### RHOSTS (required)
+
+The target host(s), see https://docs.metasploit.com/docs/using-metasploit/basics/using-metasploit.html
+
+### RPORT (required)
+
+The target port (TCP)
+
+### TARGET_USER (required)
+
+The username to change the password for (default: admin)
+
+### TARGET_PASSWORD
+
+The new password to set for the admin user (default: random)
+
+### APP_NAME
+
+The name of the app to upload (default: random)
+
+## Scenarios
+
+### Docker container running Splunk 9.0.4
+
+
+If the user you have access doen't have the capability `edit_user` the module will fail as shown below:
+
+```
+msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > check
```

Comment:
After running the setup steps and logging in to the web interface, I'm getting the following error (OpenSSL 3.1.3)
```
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > set RHOST 127.0.0.1
RHOST => 127.0.0.1
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > check

[-] Msf::OptionValidateError The following options failed to validate: USERNAME, PASSWORD
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > set USERNAME admin
USERNAME => admin
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > set PASSWORD password
PASSWORD => password
msf6 exploit(multi/http/splunk_privilege_escalation_cve_2023_32707) > check

[-] Exploit failed [unreachable]: OpenSSL::SSL::SSLError SSL_connect returned=1 errno=0 state=error: wrong version number
[-] 127.0.0.1:8000 - Check failed: The state could not be determined.
```
Are the options/OpenSSL version incorrect?

Reply:
The docker container doesn't provide a default SSL certificate for the web UI.

Please `set SSL false` in the module.

You can refer to this [documentation](https://splunk.github.io/docker-splunk/EXAMPLES.html#create-standalone-with-ssl-enabled) if you wish to configure SSL for the UI.